libraries/AP_IRLock/AP_IRLock_SITL_Gazebo.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * AP_IRLock_SITL.h
+ *
+ *  Created on: June 09, 2016
+ *      Author: Ian Chen
+ */
+#pragma once
+
+#include <AP_HAL/utility/Socket.h>
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+#include "IRLock.h"
+
+class AP_IRLock_SITL_Gazebo : public IRLock
+{
+public:
+    AP_IRLock_SITL_Gazebo();
+
+    // init - initialize sensor library
+    void init(int8_t bus) override;
+
+    // retrieve latest sensor data - returns true if new data is available
+    bool update() override;
+
+private:
+
+    /*
+      reply packet sent from simulator to ArduPilot
+     */
+    struct irlock_packet {
```

Comment:
Unless you're going to pass one of these around you can move it into the sole user.

Reply:
done